apps/system/js/advanced_telemetry.js

```diff
@@ -209,15 +230,32 @@
     this.collecting = false;
     this.metrics = null;
     this.deviceID = null;
+    // Is the user idle? Updated in handleEvent() based on an idle observer
+    this.idle = false;
   };
 
   AT.prototype.handleEvent = function handleEvent(e) {
     switch (e.type) {
       case HUDMETRICS:
-        if (this.collecting && navigator.onLine) {
-          this.transmit(e.detail);
+        if (this.merge) {
+          this.mergeSend(e.detail, false);
```

Comment:
So, I wound up changing things (several times before I settled on this) about a bit based off of this comment.  There is a lot of async here, so I tried to simplify it into things that always need to be done (e.g. we always need to check if there is data to be merged (whether it's a merge request or a transmit request... merging always needs to be done... even when the case where there is nothing to be merged).

So, I did separate the merging and the transmitting into two separate functions and got rid of the 'mergeSend' concept.  I agree it's cleaner this way.